homeassistant/components/modbus/base_platform.py

```diff
@@ -182,12 +182,25 @@ def __init__(self, hass: HomeAssistant, hub: ModbusHub, config: dict) -> None:
         self._data_type = config[CONF_DATA_TYPE]
         self._structure: str = config[CONF_STRUCTURE]
         self._scale = config[CONF_SCALE]
-        self._precision = config.get(CONF_PRECISION, 2 if self._scale < 1 else 0)
+        self._precision = config.get(CONF_PRECISION, 2)
         self._offset = config[CONF_OFFSET]
         self._slave_count = config.get(CONF_SLAVE_COUNT, None) or config.get(
             CONF_VIRTUAL_COUNT, 0
         )
         self._slave_size = self._count = config[CONF_COUNT]
+        self._value_is_int: bool = self._data_type in (
+            DataType.INT16,
+            DataType.INT32,
+            DataType.INT64,
+            DataType.UINT16,
+            DataType.UINT32,
+            DataType.UINT64,
+        )
+        if self._value_is_int:
+            if self._min_value:
+                self._min_value = int(round(self._min_value))
```

Comment:
```suggestion
                self._min_value = round(self._min_value)
```

Reply:
No that is a wrong suggestion....round() returns a float, and we want an integer.

Reply:
```text
Python 3.11.4 (main, Jun  7 2023, 18:32:58) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> type(round(4.2))
<class 'int'>
>>>
```

Reply:
[It only returns a float if round is used with ](https://docs.python.org/3/library/functions.html#round)

Reply:
OK, I will change it, I had problem on older python versions.

Reply:
Done